src/NodeApi/Native/JSNativeApi.Interop.cs

```diff
@@ -39,168 +39,1241 @@ public static void Initialize(nint libraryHandle = default)
+                napi_create_external_buffer = &DelayLoadStubs.napi_create_external_buffer;
+                napi_create_threadsafe_function = &DelayLoadStubs.napi_create_threadsafe_function;
+                napi_delete_async_work = &DelayLoadStubs.napi_delete_async_work;
+                napi_fatal_error = &DelayLoadStubs.napi_fatal_error;
+                napi_fatal_exception = &DelayLoadStubs.napi_fatal_exception;
+                napi_get_buffer_info = &DelayLoadStubs.napi_get_buffer_info;
+                napi_get_node_version = &DelayLoadStubs.napi_get_node_version;
+                napi_get_threadsafe_function_context =
+                    &DelayLoadStubs.napi_get_threadsafe_function_context;
+                napi_get_uv_event_loop = &DelayLoadStubs.napi_get_uv_event_loop;
+                napi_is_buffer = &DelayLoadStubs.napi_is_buffer;
+                napi_make_callback = &DelayLoadStubs.napi_make_callback;
+                napi_module_register = &DelayLoadStubs.napi_module_register;
+                napi_open_callback_scope = &DelayLoadStubs.napi_open_callback_scope;
+                napi_queue_async_work = &DelayLoadStubs.napi_queue_async_work;
+                napi_ref_threadsafe_function = &DelayLoadStubs.napi_ref_threadsafe_function;
+                napi_release_threadsafe_function = &DelayLoadStubs.napi_release_threadsafe_function;
+                napi_remove_async_cleanup_hook = &DelayLoadStubs.napi_remove_async_cleanup_hook;
+                napi_remove_env_cleanup_hook = &DelayLoadStubs.napi_remove_env_cleanup_hook;
+                napi_unref_threadsafe_function = &DelayLoadStubs.napi_unref_threadsafe_function;
+                node_api_get_module_file_name = &DelayLoadStubs.node_api_get_module_file_name;
+
+                //----------------------------------------------------------------------------------
+                // Embedding APIs
+                //----------------------------------------------------------------------------------
+
+                napi_await_promise = &DelayLoadStubs.napi_await_promise;
+                napi_create_environment = &DelayLoadStubs.napi_create_environment;
+                napi_create_platform = &DelayLoadStubs.napi_create_platform;
+                napi_destroy_environment = &DelayLoadStubs.napi_destroy_environment;
+                napi_destroy_platform = &DelayLoadStubs.napi_destroy_platform;
+                napi_run_environment = &DelayLoadStubs.napi_run_environment;
+#else
+                //----------------------------------------------------------------------------------
+                // js_native_api.h APIs
+                //----------------------------------------------------------------------------------
+
+                napi_add_finalizer = (delegate* unmanaged[Cdecl]<
+                    napi_env, napi_value, nint, napi_finalize, nint, nint, napi_status>)
+                    GetFunctionPointerForDelegateAndRootIt<DelegateTypes.napi_add_finalizer>(
```

Comment:
Can the explicit type arg be omitted from all these calls to `GetFunctionPointerForDelegateAndRootIt()`? It looks like it can be implicit.

Reply:
I does not work. It seems that if I omit the explicit delegate type, then the compiler uses Func<...> specialization and function wilI fail because GetFunctionPointerForDelegate does not like generic delegates.
I will add a comment about it.